src/app/api/utils/email.tsx

```diff
@@ -54,3 +61,45 @@ export async function sendVerificationEmail(
     ),
   );
 }
+
+export async function generateUnsubscribeLinkForSubscriber(
```

Comment:
@rhelmer changed this function to always generate a fresh token and update the db (upsert a new row in the new table for subscriber)